pkg/apis/cloudcredential/v1/types_azure.go

```diff
@@ -47,6 +47,24 @@ type AzureProviderSpec struct {
 	// and RoleBindings.
 	// +optional
 	DataPermissions []string `json:"dataPermissions,omitempty"`
+
+	// The following fields are required only for STS.
```

Comment:
nit: STS is Amazon lingo
```suggestion
	// The following fields are only required for Azure Workload Identity.
```